packages/ui-causal-graph-editor/src/shared/serializer.tsx

```diff
@@ -192,6 +174,28 @@ export function causalGraphSerializer(state: GraphState): CausalGraph {
         {}
     );
 
+    const edges: CausalGraph['edges'] = state.graph.reduceEdges(
+        (acc: CausalGraph['edges'], id: string, attributes: SimulationEdge, source: string, target: string) => {
+            const serializedEdge = serializeGraphEdge(attributes, nodes[source], nodes[target]);
+
+            // if the edge is backwards, we need to swap the source and target
+            if (attributes.edge_type === EdgeType.BACKWARDS_DIRECTED_EDGE) {
```

Comment:
Don't we also have to swap `destination` and `source` here?

Reply:
Good catch